OpenRA.Mods.Common/Traits/Crates/GrantExternalConditionCrateAction.cs

```diff
@@ -60,38 +60,39 @@ public override int GetSelectionShares(Actor collector)
 
 		public override void Activate(Actor collector)
 		{
-			var actorsInRange = self.World.FindActorsInCircle(self.CenterPosition, info.Range)
-				.Where(a => a != self && a != collector && a.Owner == collector.Owner && AcceptsCondition(a));
+			collector.World.AddFrameEndTask(w =>
+			{
+				if (collector.IsInWorld && !collector.IsDead)
```

Comment:
Like the `Where()` below, this should probably also check for `AcceptCondition`.

Reply:
As this isn't searching for valid actors it's an unnecessary check. The function already takes care of it.